### ISSUE TITLE
ME Interface with crafting card + Storage Bus breaks terminals

### ISSUE THREAD

**Issue by Imsvale** [AppliedEnergistics/Applied-Energistics-2#5151]
**To Reproduce**
Have an ME Interface with a crafting card, some number of some item (possibly craftable by the network) stored in it and a Storage Bus attached. Somehow, over time if not immediately, this seems to confuse the terminals and their ability to display the items stored in the network. I assume the issue arises as the item stored in the interface is consumed by other processes, or maybe just withdrawn by the player, and new items are crafted -- or simply accessed/viewed/indexed from a terminal. I'm highly unsure exactly what makes this bug tick.

**Describe the bug**
Terminals stop showing stored items, or seemingly remove all items of a type when a smaller amount is deposited or withdrawn. The items are still all there on the drives, this is only a display bug on the terminals. Once triggered, all terminals are affected.

Can be reset (temporarily) by toggling power to the storage drives. If successfully reset and items are once again shown, terminals also bug out on taking items out of or putting items into the network. E.g. having 20,000 cobblestone and taking out a stack, or any arbitrary amount, makes all the cobblestone disappear from the terminal. Again it's still there in the system, it just doesn't show in the terminal anymore. On attempting to reset, if toggling the power too quickly (I use a power toggle for my ME Drive containing all my storage drives), the terminal will only flash the items for a moment before they disappear again. The power should be off for several seconds, if not upwards of 20-30 seconds or even more.

So far the only permanent fix I discovered was when my network ran out of fuel for its energy production, and was offline for a (real world) day or two. When I came back to it, the issue was completely gone (for weeks) until I made the poor decision to try a Storage Bus attached to an ME Interface again. I'm guessing at some point the internal handling of the items, and whatever the described interface + storage bus combo does to them, goes away after some (offline?) time. For those who know how this is handled internally, this will hopefully give a clue as to how the problem arises.

I reference https://github.com/AppliedEnergistics/Applied-Energistics-2/issues/1189 which to me seems to describe the same problem.

**Expected behavior**
The terminal not to break... And being able to use the nifty interface + storage bus trick would be great.

**Additional context**

![2021-04-13_18 46 20](https://user-images.githubusercontent.com/16666305/114597850-89249880-9c91-11eb-88cd-b3f652fff94b.png)
Blank terminal. There are many, many thousands of items stored in the network, power is on and everything is fine except for this bug. The many craftable items are also not shown.

**Environment**

All details in forced crash report: https://gist.github.com/Imsvale/fe5400706abc399535518f65675e522c

**Comment by shartte** [comment]
Can you please enable debug logging in the AE config and check if your logfile says anything?

**Comment by Imsvale** [comment]
Will do. I just have to coax out the bug again, as I got it fixed through some combination of leaving the system offline and restarting the server. Not too worried about it bugging out on me anymore, now that I feel reasonably confident I can fix it again.

**Comment by Mars1984Upilami** [comment]
Actually I dont know if this is the right place or if I should open new Post or if this problem is only my own fault. But I cant see any stored Items in my ME system either. I only have 1 ME Controller, 1 ME Drive with 1 16K Cell, 1 Crafting Terminal. Energy is not an issue since my Energy cells are fully loaded at all time. It says "Device online" on all devices. Blue lamp on my Disk. 
![2021-04-23_19 03 54](https://user-images.githubusercontent.com/83086601/115905707-c4c81b00-a466-11eb-8fde-32beb7fa7265.png)

**Comment by Imsvale** [comment]
> Actually I dont know if this is the right place or if I should open new Post or if this problem is only my own fault. But I cant see any stored Items in my ME system either. I only have 1 ME Controller, 1 ME Drive with 1 16K Cell, 1 Crafting Terminal. Energy is not an issue since my Energy cells are fully loaded at all time. It says "Device online" on all devices. Blue lamp on my Disk.
> 
> This is with AE2 Fabric Alpha 3
> 
> ![2021-04-23_19 03 54](https://user-images.githubusercontent.com/83086601/115905707-c4c81b00-a466-11eb-8fde-32beb7fa7265.png)

This shouldn't be the place for your issue unless it is specific to the Interface + Storage Bus combo described. Just open a new post for some general troubleshooting, or try the [Discord](https://discord.gg/GygKjjm) for support.

**Comment by duncanwebb** [comment]
I have the same problem, the terminals show no items and items cannot be fed into the network, either by shift clicking or by using an import bus.

After removing the me interface and me storage bus the AE2 network is then broken forever. Sometimes it works, and sometimes  it doesn't, game and machine restarts don't fix this behaviour, a game restore does.

Version 8.3.0-alpha3

**Comment by Mars1984Upilami** [comment]
Hey, u wont find any help here. The case got closed afaik. 

I turned my back on fabric and went back to forge since I have no problems there. I could load my old world and now have the nice things I wanted as I wanted them to function. Had to change some things and rework my whole energy system but hey, at least I van now work with ME

**Comment by Technici4n** [comment]
> Can you please enable debug logging in the AE config and check if your logfile says anything?

**Comment by Imsvale** [comment]
For what it's worth, I haven't been able to bring out this bug again. Maybe there was something specific about my system previously that made it happen, and it has since been changed. I don't know. I've played very little lately (interest has tapered off), so I haven't had much time to play around with it. I've enabled debug logging, and if I do encounter it again, I will post the log. If not, hopefully someone else who is currently experiencing the same problem, can be of more help.

**Comment by duncanwebb** [comment]
I can consistently break AE2, once a ME interface and storage bus have been added to the network. However, the behaviour changes a bit. Sometimes the terminals are blank and sometime import and export don't work. Here is a short video showing items not being imported even though there is an empty 4k storage card in the disk drive.

https://user-images.githubusercontent.com/4198031/121057896-0ff08000-c7c0-11eb-9c03-7aa1e4ce0f29.mp4

I'm playing All of Fabric 3 and have the same problem with version 2.7.4 (appliedenergistics2-fabric-8.3.0-alpha.2) and with version 2.8.2 (appliedenergistics2-fabric-8.3.0-alpha.3)

Debugging was enabled before starting the game
config\appliedenergistics2\common.json
```
    "Misc": {
      "UnsupportedDeveloperTools": false,
      "IntegrationLogging": false,
      "WebsiteRecipes": false,
      "LogSecurityAudits": false,
      "UpdateLogging": false,
      "PacketLogging": false,
      "CraftingLog": false,
      "DebugLogging": true
    },
```
Is there a separate log file or does it go into lastest.log? I see no relevant log entries there.

**Comment by duncanwebb** [comment]
Here are the steps so that the terminal is empty.

Create a new crafting recipe and add it to an interface in the crafting cube.
That's it.

![image](https://user-images.githubusercontent.com/4198031/121192327-20aaff80-c86d-11eb-8dcf-993fb79eefb7.png)

**Comment by yorii** [comment]
Duplicate issue: #4839

**Comment by Technici4n** [comment]
Fabric 1.16 is not supported anymore, this is probably fixed in 1.17.
